packages/next-on-pages/src/buildApplication/processVercelFunctions/prerenderFunctions.ts

```diff
@@ -0,0 +1,199 @@
+}
+
+/**
+ * Retrieves the path to the prerendered route, if it exists.
+ *
+ * @param config Fallback file configuration.
+ * @param relativePath Relative path to the prerendered function's directory.
+ * @param opts Options for processing Vercel functions.
+ * @returns The path to the prerendered route, or null if it does not exist.
+ */
+async function getRoutePath(
+	{ fallback }: VercelPrerenderConfig,
+	relativePath: string,
+	{ functionsDir }: ProcessVercelFunctionsOpts,
+): Promise<string | null> {
+	const prerenderRoute = join(dirname(relativePath), fallback.fsPath);
+	const prerenderFile = join(functionsDir, prerenderRoute);
+
+	// Check the prerendered file exists.
+	if (!(await validateFile(prerenderFile))) {
+		const normalizedPath = normalizePath(prerenderRoute);
+		cliWarn(`Could not find prerendered file for ${normalizedPath}`);
+		return null;
+	}
+
+	return prerenderFile;
+}
+
+/**
+ * Retrieves the new destination for the prerendered file, if no file already exists.
+ *
+ * @example
+ * ```ts
+ * // index.prerender-fallback.html -> index.html
+ * // index.rsc.prerender-fallback.rsc -> index.rsc
+ * // favicon.ico.prerender-fallback.body -> favicon.ico
+ * // data.json.prerender-fallback.json -> data.json
+ * ```
+ *
+ * @param config Fallback file configuration.
+ * @param relativePath Relative path to the prerendered function's directory.
+ * @param opts Options for processing Vercel functions.
+ * @returns The new destination for the prerendered file, or null if a file already exists.
+ */
+async function getRouteDest(
+	{ fallback }: VercelPrerenderConfig,
+	relativePath: string,
+	{ outputDir }: ProcessVercelFunctionsOpts,
+): Promise<{ destFile: string; destRoute: string }> {
+	const fixedFileName = fallback.fsPath.replace(
+		/\.prerender-fallback(?:\.(?:rsc|body|json))?/gi,
+		'',
+	);
+	const destRoute = normalizePath(join(dirname(relativePath), fixedFileName));
+	const destFile = join(outputDir, destRoute);
+
+	return { destFile, destRoute: addLeadingSlash(destRoute) };
+}
+
+/**
+ * Copies the asset to the output directory.
+ *
+ * If an asset already exists and the hash is different, it will log a warning and be overwritten.
+ *
+ * @param asset - Information about the asset's original location, name, and destination.
+ */
+export async function copyNewFiles({
+	originalFile,
+	destFile,
+	relativeName,
+}: CopyNewFilesArgs): Promise<void> {
+	const destFileHash = getFileHash(destFile);
+	if (!destFileHash) {
+		await copyFileWithDir(originalFile, destFile);
+		return;
+	}
+
+	// We already know the original file exists, so we can safely get its hash.
+	const originalFileHash = getFileHash(originalFile) as Buffer;
+
+	if (!originalFileHash.equals(destFileHash)) {
+		cliWarn(
+			`Asset with different hash exists for ${relativeName}, overwriting...`,
+		);
+		await copyFileWithDir(originalFile, destFile);
+	}
+}
+
+type CopyNewFilesArgs = {
+	originalFile: string;
+	destFile: string;
+	relativeName: string;
+};
```

Comment:
the name of the function makes it sound generic but I believe this is assets specific?

also the function name indicates that it copies multiple files but I think it only copies one, right?

also is `new` necessary? what does it refer to?

```suggestion
/**
 * Copies a give asset to a new location.
 *
 * If an asset already exists and the hash (/content) is different, it will log a warning and will overwrite the file.
 *
 * @param asset - Information about the asset's original location, name, and destination.
 */
export async function copyAssetFile({
	originalFile,
	destFile,
	relativeName,
}: CopyAssetFileArgs): Promise<void> {
	const destFileHash = getFileHash(destFile);
	if (!destFileHash) {
		await copyFileWithDir(originalFile, destFile);
		return;
	}

	// We already know the original file exists, so we can safely get its hash.
	const originalFileHash = getFileHash(originalFile) as Buffer;

	if (!originalFileHash.equals(destFileHash)) {
		cliWarn(
			`Asset with different hash exists for ${relativeName}, overwriting...`,
		);
		await copyFileWithDir(originalFile, destFile);
	}
}

type CopyAssetFileArgs = {
	originalFile: string;
	destFile: string;
	relativeName: string;
};
```

Reply:
your assumptions are correct 😅